src/phoenix/server/api/queries.py

```diff
@@ -186,28 +187,44 @@ async def node(self, id: GlobalID, info: Info[Context, None]) -> Node:
                 metadata=dataset.metadata_,
             )
         elif type_name == "DatasetExample":
+            example_rowid = node_id
+            revision_rowid = (
+                select(func.max(models.DatasetExampleRevision.id))
+                .where(models.DatasetExampleRevision.dataset_example_id == example_rowid)
+                .scalar_subquery()
+            )
             async with info.context.db() as session:
                 example_result = (
                     await session.execute(
-                        select(models.DatasetExampleRevision, models.DatasetExample)
+                        select(models.DatasetExample, models.DatasetExampleRevision)
```

Comment:
Why not put this in a child resolver and use a data loader? This sorta falls under a bit of over-fetching (though I agree it's not really...)

Reply:
Do you mean over-fetching because we grab the entire `DatasetExampleRevision`? I agree it's over-fetching if we don't need the `input`, `output`, or `metadata` if the user doesn't ask for the `revision`. We have to the do the join on the `dataset_example_revisions` table no matter what because we need to check that the example has not been deleted.